version: "3.9"
services:
  db:
    image: postgis/postgis:16-3.4
    environment:
      POSTGRES_DB: photospots
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    ports: ["5432:5432"]
    volumes:
      - "postgres_data:/var/lib/postgresql/data/"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d photospots"]
      interval: 60s
      timeout: 3s
      retries: 20

  web:
    build: ./backend
    env_file: ./backend/.env
    depends_on:
      db:
        condition: service_healthy
    ports: ["8000:8000"]
    volumes:
      - ./backend:/app
      - media:/app/media
    command: >
      bash -lc "python manage.py migrate &&
                python manage.py runserver 0.0.0.0:8000"

  bot:
    build: ./bot
    env_file: ./bot/.env
    depends_on:
      - web

volumes:
  postgres_data:
  media:
